src/docs/sphinx/dev_guide/equation_solver.rst

```diff
@@ -0,0 +1,177 @@
+algorithms needed to solve these systems of equations. Note that while some nonlinear solvers do not depend on an embedded linear solver (e.g. L-BFGS), we require a linear 
+solver to be specified as it is used to compute reasonable initial guesses and perform adjoint solves within Serac. 
+
+The key methods provided by this class are:
+
+1.  ``void SetOperator(const mfem::Operator& op)``: This defines the nonlinear ``mfem::Operator`` representing the vector-valued nonlinear system of equations :math:`F`.
+2.  ``void Solve(mfem::Vector& x)``: This solves the nonlinear system :math:`F(\mathbf{X}) = \mathbf{0}` and stores the solution in-place in ``x``.
+3.  ``mfem::Solver& LinearSolver()``: This returns the associated linear solver for adjoint solves and initial guess calculations.
+
+Two workflows exist for defining the linear and nonlinear algorithms in a ``EquationSolver``: predefined option structs for common use cases and fully custom ``mfem::Solvers``
+for advanced users.
+
+Common configurations via option structs
+----------------------------------------
+
+The easist way to build an equation solver is by providing structs containing parameters for common configurations of the linear and nonlinear solvers. They can also
+be passed directly to physics module constructors.
+
+.. literalinclude:: ../../../../src/serac/numerics/equation_solver.hpp
+   :start-after: _build_equationsolver_start
+   :end-before: _build_equationsolver_end
+   :language: C++
+
+The nonlinear solver configuration options are provided by the ``NonlinearSolverOptions`` struct:
+
+.. literalinclude:: ../../../../src/serac/numerics/solver_config.hpp
+   :start-after: _nonlinear_options_start
+   :end-before: _nonlinear_options_end
+   :language: C++
+
+The current possible nonlinear solution algorithms are:
+
+.. literalinclude:: ../../../../src/serac/numerics/solver_config.hpp
+   :start-after: _nonlinear_solvers_start
+   :end-before: _nonlinear_solvers_end
+   :language: C++
+
+The linear solver configuration options are provided by the ``LinearSolverOptions`` struct:
+
+.. literalinclude:: ../../../../src/serac/numerics/solver_config.hpp
+   :start-after: _linear_options_start
+   :end-before: _linear_options_end
+   :language: C++
+
+The current possible linear solution algorithms are:
+
+.. literalinclude:: ../../../../src/serac/numerics/solver_config.hpp
+   :start-after: _linear_solvers_start
+   :end-before: _linear_solvers_end
+   :language: C++
+
+The current possible preconditioners for iterative linear solvers are:
+
+.. literalinclude:: ../../../../src/serac/numerics/solver_config.hpp
+   :start-after: _preconditioners_start
+   :end-before: _preconditioners_end
+   :language: C++
+
+As an example, the default solid mechanics ``EquationSolver`` can be built using the following syntax:
+
+.. code-block:: cpp
+
+  const serac::NonlinearSolverOptions nonlinear_options = {.nonlin_solver  = NonlinearSolver::Newton,
```

Comment:
Is this section anywhere in the code so we can ensure that it does not get out of date?